receiver/dockerreceiver/README.md

```diff
@@ -0,0 +1,80 @@
+# Docker Receiver
+
+<!-- status autogenerated section -->
+| Status        |           |
+| ------------- |-----------|
+| Stability     | [alpha]: metrics   |
+| Distributions | [contrib] |
+| Issues        | [![Open issues](https://img.shields.io/github/issues-search/open-telemetry/opentelemetry-collector-contrib?query=is%3Aissue%20is%3Aopen%20label%3Areceiver%2Fdocker%20&label=open&color=orange&logo=opentelemetry)](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues?q=is%3Aopen+is%3Aissue+label%3Areceiver%2Fdocker) [![Closed issues](https://img.shields.io/github/issues-search/open-telemetry/opentelemetry-collector-contrib?query=is%3Aissue%20is%3Aclosed%20label%3Areceiver%2Fdocker%20&label=closed&color=blue&logo=opentelemetry)](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues?q=is%3Aclosed+is%3Aissue+label%3Areceiver%2Fdocker) |
+| [Code Owners](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#becoming-a-code-owner)    | [@rmfitzpatrick](https://www.github.com/rmfitzpatrick), [@jamesmoessis](https://www.github.com/jamesmoessis), [@aboguszewski-sumo](https://www.github.com/aboguszewski-sumo) |
+
+[alpha]: https://github.com/open-telemetry/opentelemetry-collector#alpha
+[contrib]: https://github.com/open-telemetry/opentelemetry-collector-releases/tree/main/distributions/otelcol-contrib
+<!-- end autogenerated section -->
+
+The Docker receiver queries the local Docker daemon's API to fetch telemetry.
+Currently, only Docker stats are available.
+The metrics are fetched for all desired running containers on a configured interval.
+These stats are for container
+resource usage of cpu, memory, network, and the
+[blkio controller](https://www.kernel.org/doc/Documentation/cgroup-v1/blkio-controller.txt).
+
+> :information_source: Requires Docker API version 1.22+ and only Linux is supported.
+
+## Configuration
+
+The following settings are optional and affect all telemetry:
+
+- `endpoint` (default = `unix:///var/run/docker.sock`): Address to reach the desired Docker daemon.
+- `api_version` (default = `1.25`): The Docker client API version (must be 1.25+). If using one with a terminating zero, input as a string to prevent undesired truncation (e.g. `"1.40"` instead of `1.40`, which is parsed as `1.4`). [Docker API versions](https://docs.docker.com/engine/api/).
+
+### Stats
```

Comment:
I appreciate that you thought about separating metrics collection config from logs/events collection config at this early stage.

After looking at these config properties in detail, it seems to me that at least some of these could be useful for logs collection as well:
- `timeout` could probably easily be a common setting between both metrics and logs?
- `excluded_images` - I wonder if this setting could also allow to filter Docker events about containers? Probably need to check how this is implemented in the Docker client library.
- `container_labels_to_metrics_labes` - seems like this might be usable for Docker container events as well. Not sure about `env_vars_to_metric_labes`, probably need to check if env vars are available in the Docker events.

Reply:
Let's consider the following user case: a user wants to collect metrics, excluding container `a` and events for all containers. What do we want them to do: create two separate `dockerreceiver`s or create one with different configs?

From my perspective, both are valid, but I'm not sure which one would be in the spirit of OpenTelemetry.

Reply:
This is a very good phrasing of the question :) I'm not aware of a definitive answer on "the OpenTelemetry way". I would lean towards the user specifying two different components for this. I'm definitely not in favor of having separate options in components that do (mostly) the same thing, but for a different signal: logs, metrics, traces.

Reply:
@andrzej-stencel I made a PR: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33038
let's move further discussions on this aspect there.